### PR TITLE
Jitter for RetryingTest

### LINKED ISSUE #783: RetryingTest Extension: Jitter factor
From [Jitter factor discussion](https://github.com/junit-pioneer/junit-pioneer/discussions/778#discussioncomment-7377835): Idea is to add a [Jitter](https://en.wikipedia.org/wiki/Jitter) factor for retrying tests, so the retries are not exactly executed after the same time.

### PR DESCRIPTION
**Pull request by IlyasYOY**
Proposed commit message:

```
Add (#783) maxJitterMs and jitterSeed attributes to @RetryingTest

Implement random jitter calculation in RetryingTestExtension, 
validate non‑negative jitter values, update retry messages to 
include jitter information, and extend tests to cover jitter 
behavior and related edge cases.
```

I'm concerned about the test coverage. The logic around randomness is easy, but hard to test. I didn't find any existing code to rely on. I'd be glad to write more tests if you think they are required and have idea on the good way to do so in the project.

Closes: https://github.com/junit-pioneer/junit-pioneer/issues/783

---
**PR checklist**

The following checklist shall help the PR's author, the reviewers and maintainers to ensure the quality of this project.
It is based on our contributors guidelines, especially the ["writing code" section](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.adoc#writing-code).
It shall help to check for completion of the listed points.
If a point does not apply to the given PR's changes, the corresponding entry can be simply marked as done. 

Documentation (general)
* [x] There is documentation (Javadoc and site documentation; added or updated)
* [x] There is implementation information to describe _why_ a non-obvious source code / solution got implemented
* [-] Site documentation has its own `.adoc` file in the `docs` folder, e.g. `docs/report-entries.adoc`
* [x] Site documentation in `.adoc` file references demo in `src/demo/java` instead of containing code blocks as text
* [x] Only one sentence per line (especially in `.adoc` files)
* [x] Javadoc uses formal style, while sites documentation may use informal style

Documentation (new extension)
* [-] The `docs/docs-nav.yml` navigation has an entry for the new extension
* [-] The `package-info.java` contains information about the new extension

Code (general)
* [x] Code adheres to code style, naming conventions etc.
* [-] Successful tests cover all changes **(?)**
* [x] There are checks which validate correct / false usage / configuration of a functionality and there are tests to verify those checks
* [x] Tests use [AssertJ](https://assertj.github.io/doc/) or our own [PioneerAssert](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.adoc#assertions) (which are based on AssertJ)

Code (new package)
* [-] The new package is exported in `module-info.java`
* [-] The new package is also present in the tests
* [-] The new package is opened for reflection to JUnit 5 in `module-info.java`
* [-] The new package is listed in the contribution guide

Contributing
* [x] A prepared commit message exists
* [x] The list of contributions inside `README.adoc` mentions the new contribution (real name optional) 

## Summary by CodeRabbit

* **New Features**
  * Added jitter to retry behavior with configurable maximum jitter and optional seed for reproducible delays.

* **Documentation**
  * Added "Configuring the Jitter" section with runnable demos showing jitter and seeded-jitter usage.

* **Tests**
  * Expanded tests and assertions covering jitter, seeded jitter, suspension timing, and retry scenarios.

### CODE REVIEW

**Comment by IlyasYOY** [comment]
@Michael1993 Hello! 

Could you please take a look? I think I've fixed all AI comments.

**Comment by IlyasYOY** [comment]
Hello! @Michael1993 any updates on this? Could you take a look?

File: src/main/java/org/junitpioneer/jupiter/RetryingTestExtension.java [hunk 1]
```diff
@@ -86,11 +89,15 @@ private static class FailedTestRetrier implements Iterator<RetryingTestInvocatio
...
 		private boolean seenUnexpectedException;
 
-		private FailedTestRetrier(int maxRetries, int minSuccess, int suspendForMs,
+		private FailedTestRetrier(int maxRetries, int minSuccess, int suspendForMs, int maxJitterMs, int jitterSeed,
 				Class<? extends Throwable>[] expectedExceptions, TestNameFormatter formatter) {
 			this.maxRetries = maxRetries;
 			this.minSuccess = minSuccess;
 			this.suspendForMs = suspendForMs;
+			this.maxJitterMs = maxJitterMs;
+			// TODO: Shouldn't this value be logged so the user could reproduce the tests?
```
[the remark this answers is not in the document]
Comment (Michael1993) [reply src/main/java/org/junitpioneer/jupiter/RetryingTestExtension.java:98]: Should be plainly visible in the `@RetryingTest` annotation.

**Review by Michael1993 — APPROVED** [review]

**Comment by Michael1993** [comment]
Would you like to update the README to add yourself as a contributor?

**Review by Bukama — APPROVED** [review]

**Comment by IlyasYOY** [comment]
@Michael1993 
no, thanks for asking 🙏 

I've been there [once](https://github.com/junit-pioneer/junit-pioneer/blob/main/README.adoc#2023), I think that's enough 😅
